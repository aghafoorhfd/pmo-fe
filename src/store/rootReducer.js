import { combineReducers } from 'redux';
import theme from './slices/themeSlice';
import auth from './slices/authSlice';
import user from './slices/userSlice';
import resourceTeam from './slices/resourceTeamSlice';
import projectDetails from './slices/projectDetailsSlice';
import riskManagement from './slices/riskManagementSlice';
import analytics from './slices/analyticsSlice';
import customerActivation from './slices/customerActivationSlice';
import filters from './slices/filterSlice';
import jira from './slices/jiraSlice';
import subscription from './slices/subscriptionSlice';

const rootReducer = (asyncReducers) => (state, action) => {
  const combinedReducer = combineReducers({
    auth,
    riskManagement,
    projectDetails,
    resourceTeam,
    theme,
    user,
    analytics,
    customerActivation,
    filters,
    jira,
    subscription,
    ...asyncReducers
  });
  return combinedReducer(state, action);
};

export default rootReducer;
