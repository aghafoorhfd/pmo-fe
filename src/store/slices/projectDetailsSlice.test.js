import store from '../index';

describe('Project Details Slice', () => {
  it('should show the initial state for projectDetails', () => {
    const state = store.getState().projectDetails;
    expect(state).toStrictEqual({
      loading: false,
      message: '',
      showMessage: false,
      configurationMetrics: {},
      generalUsers: [],
      status: 'success',
      projectList: [],
      selectedProjectDetails: ''
    });
    expect(state).toMatchSnapshot();
  });
});
