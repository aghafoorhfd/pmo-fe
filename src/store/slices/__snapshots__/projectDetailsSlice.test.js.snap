// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Details Slice should show the initial state for projectDetails 1`] = `
{
  "configurationMetrics": {},
  "generalUsers": [],
  "loading": false,
  "message": "",
  "projectList": [],
  "selectedProjectDetails": "",
  "showMessage": false,
  "status": "success",
}
`;
